src/components/UI/FormInput.vue

```diff
@@ -0,0 +1,111 @@
+<script lang="ts" setup>
+import { computed } from 'vue'
+import Icon from '/@/components/UI/Icon.vue'
+
+interface Props {
+  modelValue: string
+  placeholder?: string
+  limit?: number
+  hasAtmark?: boolean
+  hasAnchor?: boolean
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  placeholder: '',
+  limit: undefined,
+  hasAtmark: false,
+  hasAnchor: false
+})
+const emit = defineEmits<{
+  (e: 'update:modelValue', modelValue: string): void
+}>()
+
+const isExceeded = computed(
+  () => props.limit && props.modelValue.length > props.limit
+)
+const isInvalidLink = computed(
+  () => props.hasAnchor && !props.modelValue.startsWith('http')
```

Comment:
```suggestion
  () => props.hasAnchor && !/^https?:/.test(props.modelValue))
```
簡易的な検査を意図していても`httpaa`ははじいたほうがよさそうです

(ちゃんとやるなら`new URL`やって`catch`したかをチェックする)

Reply:
なるほどです
ちゃんとやってみたい感があるので調べてみます👍